codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordDecoder.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.dns;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.handler.codec.CorruptedFrameException;
+import io.netty.util.CharsetUtil;
+import io.netty.util.internal.StringUtil;
+
+/**
+ * The default {@link DnsRecordDecoder} implementation.
+ *
+ * @see DefaultDnsRecordEncoder
+ */
+public class DefaultDnsRecordDecoder implements DnsRecordDecoder {
+
+    /**
+     * Creates a new instance.
+     */
+    protected DefaultDnsRecordDecoder() { }
```

Comment:
Yes. The default instance is provided by `DnsRecordDecoder.DEFAULT`, so there's no reason for a user to instantiate it.  However, a user can extend it to override the default behavior.